app/models/game.rb

```diff
@@ -14,6 +14,30 @@ def populate!
     populate_right_white_half!
   end
 
+  def white_check?
```

Comment:
Ok, I managed to fit it in line limit but I still have an ABC error in rubocop [15.3/15]. It is only *0.3* points away from being Rubocop compliant.

```Ruby
def check?(color)
    if color == 'white'
      blk_pieces.each do |p|
        return true if p.valid_move?(w_k.x_position, w_k.y_position)
      end
    else
      wht_pieces.each do |p|
        return true if p.valid_move?(b_k.x_position, b_k.y_position)
      end
    end
    false
  end

  private

  def b_k
    pieces.find_by(type: 'King', color: 'black')
  end

  def w_k
    pieces.find_by(type: 'King', color: 'white')
  end

  def wht_pieces
    pieces.select { |p| p.color = 'white' }
  end

  def blk_pieces
    pieces.select { |p| p.color = 'black' }
  end
```

Here is the reference for ABC: http://c2.com/cgi/wiki?AbcMetric

Firstly, I know having `b_k` and `w_k` variables for Kings is not great, but that was the only way I could fit code in one line.

Secondly, I agree, having a color param is better (it's just I couldn't fit the whole method in before). Now, I can actually use this logic to prevent each Piece to open a check to their own king with this in our `move!` method:

`return false if x_position = x && y_position = y && check?(color)`

I need to help to refactor this code to make it work.

I got the checkmate logic all written up too (no PR yet), just need to finish this up and then carry on with checkmate and tests for it.